includes/helper/class-wp-job-manager-com-auth-token.php

```diff
@@ -0,0 +1,177 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Com_Auth_Token.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Helper functions used for creating and validating tokens used for authenticating with WPJobManager.com.
+ *
+ * @package wp-job-manager
+ * @since   $$next-version$$
+ */
+class WP_Job_Manager_Com_Auth_Token {
+	/**
+	 * The meta key used to store the token.
+	 */
+	const META_KEY = 'wpjmcom_site_auth_token';
+
+	/**
+	 * The accepted object types to be associated with the token.
+	 */
+	const ACCEPTED_OBJECT_TYPES = [ 'post', 'user' ];
+
+	/**
+	 * The expiration time for the token, in seconds.
+	 */
+	const EXPIRATION_TIME = MINUTE_IN_SECONDS;
+
+	/**
+	 * The singleton instance of the class.
+	 *
+	 * @var WP_Job_Manager_Com_Auth_Token
+	 */
+	private static $instance;
+
+	/**
+	 * WP_Job_Manager_Com_Auth_Token constructor.
+	 */
+	private function __construct() {}
+
+	/**
+	 * Returns the singleton instance of the class.
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( null === self::$instance ) {
+			self::$instance = new WP_Job_Manager_Com_Auth_Token();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Generates the site token associated with an object type and object id
+	 *
+	 * @param string $object_type Type of the object associated with the token. Accepts 'post' or 'user'.
+	 * @param int    $object_id The ID of the object to associate the token with.
+	 * @return string|WP_Error The token generated, or a WP_Error if the token could not be generated/persisted
+	 */
+	public function generate( $object_type, $object_id ) {
+		if ( ! in_array( $object_type, self::ACCEPTED_OBJECT_TYPES, true ) ) {
+			return new WP_Error( 'wpjobmanager-com-invalid-type', __( 'Invalid object type', 'wp-job-manager' ) );
+		}
+		$token = $this->generate_new_token();
+		if ( is_wp_error( $token ) ) {
+			return $token;
+		}
+		$encoded = $this->encode( $token );
+		$result  = add_metadata( $object_type, $object_id, self::META_KEY, $encoded );
+		if ( ! $result ) {
+			return new WP_Error( 'wpjobmanager-com-token-not-saved', __( 'Token could not be persisted', 'wp-job-manager' ) );
+		}
+		return $token;
+	}
+
+	/**
+	 * Generates a new random token and return it
+	 *
+	 * @return string|WP_Error The random token generated, or a WP_Error if the token could not be generated.
+	 */
+	private function generate_new_token() {
```

Comment:
This method will always generate a string with 64 characters, from 0 to 9 and a to f (hexadecimal). Totally random.